x/fswap/types/codec.go

```diff
@@ -2,22 +2,48 @@ package types
 
 import (
 	"github.com/Finschia/finschia-sdk/codec"
-	cdctypes "github.com/Finschia/finschia-sdk/codec/types"
-	// this line is used by starport scaffolding # 1
+	"github.com/Finschia/finschia-sdk/codec/types"
+	cryptocodec "github.com/Finschia/finschia-sdk/crypto/codec"
+	sdk "github.com/Finschia/finschia-sdk/types"
 	"github.com/Finschia/finschia-sdk/types/msgservice"
+	authzcodec "github.com/Finschia/finschia-sdk/x/authz/codec"
+	fdncodec "github.com/Finschia/finschia-sdk/x/foundation/codec"
+	govcodec "github.com/Finschia/finschia-sdk/x/gov/codec"
+	govtypes "github.com/Finschia/finschia-sdk/x/gov/types"
 )
 
 func RegisterCodec(cdc *codec.LegacyAmino) {
 	// this line is used by starport scaffolding # 2
 }
 
-func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&MakeSwapProposal{}, "finschia-sdk/MakeSwapProposal", nil)
```

Comment:
There is no legacy.RegisterAminoMsg for `MsgSwap` and `MsgSwapAll`. Please check it.